app/src/androidTest/java/com/amaze/filemanager/asynchronous/services/ftp/FTPServiceEspressoTest.java

```diff
@@ -83,6 +101,89 @@ private void loginAndVerifyWith(FTPClient ftpClient) throws Exception
             fail("Download folder not found on device. Either storage is not available, or something is really wrong with FTPService. Check logcat.");
     }
 
+    private void testUploadWith(FTPClient ftpClient) throws Exception
```

Comment:
I think it is better to throw something specific and let the app crash if something completely unexpected happens.

Reply:
Though I used an idiot way for assertion (`fail()` instead of `assertTrue()` which is corrected now), I think the assertion needs to be here, and failures accessing the filesystem will crash the service and the test anyway.

Reply:
Sorry, I'm not following you, I meant to say that instead of `throws Exception` you should put a specific subclass, like `throws IOException` for example.